src/Hit/Git.hs

```diff
@@ -60,17 +60,38 @@ runNew issueNum = do
         . T.filter (\c -> isAlphaNum c || isDigit c || isSpace c)
 
 -- | @hit commit@ command.
-runCommit :: Text -> Bool -> IO ()
-runCommit (T.strip -> msg) (not -> hasIssue)
-    | msg == "" = errorMessage "Commit message cannot be empty"
-    | otherwise = do
-        branch <- getCurrentBranch
-        let issueNum = issueFromBranch branch
-        "git" ["add", "."]
-        "git" ["commit", "-m", showMsg $ guard hasIssue *> issueNum]
+runCommit :: Maybe Text -> Bool -> IO ()
+runCommit maybeMsg (not -> hasIssue) = case maybeMsg of
+    Just (T.strip -> msg) ->
+        if msg == ""
+        then errorMessage "Commit message cannot be empty"
+        else do
+            issueNum <- getCurrentIssue
+            commitCmds msg issueNum
+    {- if the commit name is not specified then check the branchName
+    If this is issue-related branch, take the issue name as the commit name.
+    Otherwise print errorMessage.
+    -}
+    Nothing -> do
+        issueNum <- getCurrentIssue
+        case issueNum of
+            Nothing -> errorMessage "Commit message cannot be empty: can not be taken from the context"
+            Just n -> do
+                let issueId = mkIssueId n
+                -- TODO: which symbols are not allowed?
+                title <- getIssueTitle issueId
+                commitCmds title issueNum
```

Comment:
Can't think of a better way to rewrite this piece of code... I can read it and understand what's going on here. And it's still quite short. Unfortunately, we have to do all these pattern-matching to do case-analysis.

One way that can potentially improve code readability is to introduce helper sum type as a in-between layer. So we analyse inputs and return one of the constructors and then pattern-match on those constructors to decide what to do. This helps only if a function has a lot of non-trivia if-then-else or very nested pattern-matching. Not sure it will help us here. But here is the idea itself:

```haskell
data CommitEvent
    = EmptyMessage
    | NoIssueNumber
    | CommitMessage Text
    | FromIssueTitle Int

mkCommitEvent :: Maybe Text -> IO CommitEvent

runCommit = mkCommitEvent maybeMsg >>= \case
```

This is just an idea. It can help in big cases and big functions. Not sure it will help us here.

Reply:
Hmm, I think it's okay without this data type now... I'm not sure if it worth it...

Reply:
@vrom911 I tend to this so as well. I think the data type only makes sense when the logic becomes very difficult and nested with a lot of cases. But we don't have that many cases for now.